### DISCUSSION TITLE
What happen if I try to find a document while database connection is not fully ready ?

### DISCUSSION THREAD

**Discussion by Apokalypt** [Automattic/mongoose#14907] [Q&A] [answered]
### Context
I've developed an API hosted on a service that stops if there's no call after a certain amount of time and starts as soon as a new request arrives. Currently, when the service starts, I connect to my database with this code:
`mongoose.connect(process.env.DATABASE_URL_CONNECTION)`
However, connecting to the database can take some time, and in some cases my API refuses the HTTP call because the connection is not ready (`mongoose.connection.readyState !== mongoose.ConnectionStates.connected`).

### Question
So I was wondering what would happen if I removed the code that makes sure the connection is ready before accepting an HTTP request.
If the connection isn't ready and I try to list the documents in a collection using the query `<Model>.find({ ... })` what will happen? Will the library wait until the connection is ready? Will the library raise an error indicating that the connection is not ready?

**Comment by vkarpov15** [comment, marked as answer]
Presuming you already called mongoose.connect(), the MongoDB node driver will automatically retry any queries for up to serverSelectionTimeoutMS (30 seconds by default). So your find() call will wait until mongoose.connect() succeeds up to a certain timeout. If it takes more than 30 seconds to connect, your find() will error out
